velox/common/base/BitUtil.cpp

```diff
@@ -118,13 +118,16 @@ void scatterBits(
     return;
   }
 #ifdef __BMI2__
-  int32_t highByte = numTarget / 8;
-  int32_t highBit = numTarget & 7;
-  int lowByte = std::max(0, highByte - 7);
+  uint32_t highByte = numTarget / 8;
```

Comment:
You can `static_cast<uint32_t>(numTarget)` if we keep `numTarget` signed to still get perf gain here.